src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs

```diff
@@ -233,10 +242,25 @@ internal IEnumerable<string> CreateArgument()
             {
                 foreach (var arg in this.VSTestCollect)
                 {
+                    if (arg.Equals("Code Coverage", StringComparison.OrdinalIgnoreCase))
+                    {
+                        isCollectCodeCoverageEnabled = true;
+                    }
                     allArgs.Add("--collect:" + ArgumentEscaper.HandleEscapeSequenceInArgForProcessStart(arg));
                 }
             }
 
+            if (isCollectCodeCoverageEnabled || isRunSettingsEnabled)
+            {
+                // Pass TraceDataCollector path to vstest.console as TestAdapterPath if --collect "Code Coverage"
```

Comment:
Here trade off is b/w always parsing runsettings(which we don't want to do), vs always adding the path of TraceDataCollector dir if it is present, & runsettings is used.

I agree with latter.

Reply:
Done.